lib/config/fileconf_test.go

```diff
@@ -402,20 +394,82 @@ func TestAuthenticationSection(t *testing.T) {
 					},
 				}
 			},
-			expectError: require.NoError,
 			expected: &AuthenticationConfig{
 				DeviceTrust: &DeviceTrust{
 					Mode:       "required",
 					AutoEnroll: "yes",
 				},
 			},
+		}, {
+			desc: "signature suite empty",
+			mutate: func(cfg cfgMap) {
+				cfg["auth_service"].(cfgMap)["authentication"] = cfgMap{
+					"signature_algorithm_suite": "",
+				}
+			},
+			expected: &AuthenticationConfig{
+				SignatureAlgorithmSuite: types.SignatureAlgorithmSuite_SIGNATURE_ALGORITHM_SUITE_UNSPECIFIED,
+			},
+		}, {
+			desc: "signature suite legacy",
+			mutate: func(cfg cfgMap) {
+				cfg["auth_service"].(cfgMap)["authentication"] = cfgMap{
+					"signature_algorithm_suite": "legacy",
+				}
+			},
+			expected: &AuthenticationConfig{
+				SignatureAlgorithmSuite: types.SignatureAlgorithmSuite_SIGNATURE_ALGORITHM_SUITE_LEGACY,
+			},
+		}, {
+			desc: "signature suite balanced-v1",
+			mutate: func(cfg cfgMap) {
+				cfg["auth_service"].(cfgMap)["authentication"] = cfgMap{
+					"signature_algorithm_suite": "balanced-v1",
+				}
+			},
+			expected: &AuthenticationConfig{
+				SignatureAlgorithmSuite: types.SignatureAlgorithmSuite_SIGNATURE_ALGORITHM_SUITE_BALANCED_V1,
+			},
+		}, {
+			desc: "signature suite fips-v1",
+			mutate: func(cfg cfgMap) {
+				cfg["auth_service"].(cfgMap)["authentication"] = cfgMap{
+					"signature_algorithm_suite": "fips-v1",
+				}
+			},
+			expected: &AuthenticationConfig{
+				SignatureAlgorithmSuite: types.SignatureAlgorithmSuite_SIGNATURE_ALGORITHM_SUITE_FIPS_V1,
+			},
+		}, {
+			desc: "signature suite hsm-v1",
+			mutate: func(cfg cfgMap) {
+				cfg["auth_service"].(cfgMap)["authentication"] = cfgMap{
+					"signature_algorithm_suite": "hsm-v1",
+				}
+			},
+			expected: &AuthenticationConfig{
+				SignatureAlgorithmSuite: types.SignatureAlgorithmSuite_SIGNATURE_ALGORITHM_SUITE_HSM_V1,
+			},
+		}, {
+			desc: "signature suite typo",
+			mutate: func(cfg cfgMap) {
+				cfg["auth_service"].(cfgMap)["authentication"] = cfgMap{
+					"signature_algorithm_suite": "balanced-v0",
+				}
+			},
+			expectError: func(t require.TestingT, err error, msgAndArgs ...interface{}) {
+				require.ErrorContains(t, err, "invalid value: balanced-v0")
+			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.desc, func(t *testing.T) {
 			text := bytes.NewBuffer(editConfig(t, tt.mutate))
 			cfg, err := ReadConfig(text)
-			tt.expectError(t, err)
+			if tt.expectError != nil {
+				tt.expectError(t, err)
+				return
+			}
 
```

Comment:
when expectError is nil it should require.NoError

Reply:
good catch thanks